pkg/cluster/resources.go

```diff
@@ -135,18 +135,14 @@ func CreateNamespace(cli client.Client, namespace string, metaOptions ...MetaOpt
 	}
 
 	foundNamespace := &corev1.Namespace{}
-	err := cli.Get(context.TODO(), client.ObjectKey{Name: namespace}, foundNamespace)
-	if err != nil {
-		if apierrs.IsNotFound(err) {
-			err = cli.Create(context.TODO(), desiredNamespace)
-			if err != nil && !apierrs.IsAlreadyExists(err) {
-				return nil, err
-			}
-			desiredNamespace.DeepCopyInto(foundNamespace)
-		} else {
-			return nil, err
-		}
+	if getErr := cli.Get(context.TODO(), client.ObjectKey{Name: namespace}, foundNamespace); client.IgnoreNotFound(getErr) != nil {
```

Comment:
Wondering, why you need different variables for errors. Self-documenting code with error types?

Reply:
Yeah, a habit to minimize shadowing of `err`, though I would expect linter is catching it, but I am not sure if our config enables this particular linter